src/common/res/features/check-credit-balances/main.js

```diff
@@ -1,64 +1,153 @@
+            ynabToolKit.checkCreditBalances.updateRow(a.name);
+            ynabToolKit.checkCreditBalances.updateInspectorStyle(a.name);
           }
         });
-      },
+      };
 
-      this.observe = function(changedNodes) {
+      this.updateRow = function(name) {
+        var rows = $('.is-sub-category.is-debt-payment-category');
+        rows.each(function(i) {
+          var accountName = $(this)
+            .find('.budget-table-cell-name div.button-truncate').prop('title');
+          if (name === accountName) {
+            var categoryBalance = $(this)
+              .find('.budget-table-cell-available-div .user-data.currency');
+            categoryBalance.removeClass('positive negative')
+              .addClass('cautious');
+          }
+        });
+      };
 
-        if (changedNodes.has('budget-inspector')) {
-          // The user has changed their budget row selection
-          ynabToolKit.checkCreditBalances.invoke();
+      this.updateInspectorStyle = function(name) {
+        var inspectorName = $('.inspector-category-name.user-data').text().trim();
+        if (name === inspectorName) {
+          var inspectorBalance = $('.inspector-overview-available .user-data .user-data.currency');
+          inspectorBalance.removeClass('positive negative')
+            .addClass('cautious');
         }
       };
 
+      this.updateInspectorButton = function(name, difference) {
+        var inspectorName = $('.inspector-category-name.user-data').text().trim();
 
+        if (name && name === inspectorName) {
 
-    };
-    ynabToolKit.checkCreditBalances.invoke(); // Run itself once
+          if ($('.rectify-difference').length)
+            return;
 
-  } else {
-   setTimeout(poll, 250);
-  }
-})();
+          var fhDifference = ynabToolKit.shared.formatCurrency(difference, true);
+          var fDifference = ynabToolKit.shared.formatCurrency(difference, false);
+          var button = ' \
+          <button class="budget-inspector-button rectify-difference" \
+            onClick="ynabToolKit.checkCreditBalances.updateCreditBalances(\'' + name+'\', ' + difference + ')"> \
+            Rectify Available for PIF CC: \
+              <strong class="user-data" title="' + fDifference + '"> \
+                <span class="user-data currency zero"> \
+                ' + fhDifference + ' \
+              </span> \
+            </strong> \
+          </button>';
 
+          $('.inspector-quick-budget .ember-view').append(button);
+        }
+      };
 
+      this.updateCreditBalances = function(name, difference) {
+        var debtPaymentCategories = $('.is-debt-payment-category.is-sub-category');
 
+        $(debtPaymentCategories).each(function() {
+          var accountName = $(this).find('.budget-table-cell-name div.button-truncate').prop('title');
+          if (accountName === name) {
+            var input = $(this).find('.budget-table-cell-budgeted div.currency-input').click().find('input');
```

Comment:
There's probably a way to pull this from the view models for the actual calculation, but since stuffing it back in should happen with the input in case YNAB want to do anything with it, I say all good!